rust/lance/src/dataset/scanner.rs

```diff
@@ -1584,6 +1584,96 @@ impl Scanner {
         Ok(plan)
     }
 
+    async fn fragments_covered_by_fts_leaf(
+        &self,
+        column: &str,
+        accum: &mut RoaringBitmap,
+    ) -> Result<bool> {
+        let index = self
+            .dataset
+            .load_scalar_index_for_column(column)
```

Comment:
is this operation expensive if the helper calls it multiple times?

Reply:
Only if the index cache is completely disabled but we already call it a few times so caching the results after the first call is a bit of an existing "larger optimization"

Reply:
ok thanks, figured that was probably the case.